commons/src/main/java/org/codice/ddf/admin/commons/requests/RequestUtils.java

```diff
@@ -120,58 +127,38 @@ public ProbeReport sendGetRequest(String url, String userName, String password)
      * @param content - Body of the post request
      * @return report
      */
-    public ProbeReport sendPostRequest(String url, String username, String password,
-            String contentType, String content) {
+    public ProbeReport sendPostRequest(String url, String username, String password, String contentType, String content,
+                                       SecurityManager securityManager) {
         Report connectReport = endpointIsReachable(url);
         if (connectReport.containsFailureMessages()) {
             return new ProbeReport(connectReport.messages());
         }
-        HttpPost post = new HttpPost(url);
-        post.setHeader("Content-Type", contentType);
-        if (url.startsWith("https") && username != null && password != null) {
-            String auth = Base64.getEncoder().encodeToString((username + ":" + password).getBytes());
-            post.setHeader(HttpHeaders.AUTHORIZATION, "Basic " + auth);
+
+        Optional<WebClient> c = generateClient(url, username, password, securityManager);
+        if (!c.isPresent()) {
+            return new ProbeReport().addMessage(createRequestConfigMsg(CANNOT_CONNECT));
         }
-        try {
-            post.setEntity(new StringEntity(content));
-        } catch (UnsupportedEncodingException e) {
-            return new ProbeReport(ConfigurationMessage.createInternalErrorMsg("Error encoding query."));
+        WebClient client = c.get();
+
+        Response r = client.type(contentType).post(content);
+        if (r.getStatus() == HTTP_OK) {
+            return new ProbeReport().addMessage(createRequestConfigMsg(EXECUTED_REQUEST))
+                    .probeResults(responseToMap(r));
         }
-        return sendHttpRequest(post);
+        return new ProbeReport().addMessage(createRequestConfigMsg(CANNOT_CONNECT));
     }
 
-    protected ProbeReport sendHttpRequest(HttpRequestBase request) {
-        ProbeReport results = new ProbeReport();
-        CloseableHttpClient client = null;
-        CloseableHttpResponse response = null;
+    private Optional<WebClient> generateClient(String url, String username, String password, SecurityManager securityManager) {
+        clientFactory = (username == null && password == null) ?
```

Comment:
❗️ This isn't a threadsafe usage. You're at risk of simultaneous requests (very rare, admittedly, in the admin console case but _not impossible_) crossing streams and using the wrong credentials. There's no reason why `clientFactory` needs to be a field; it should be isolated to this method.